app/controllers/accounts_controller.rb

```diff
@@ -31,9 +31,28 @@ def create
     redirect_to my_account_path
   end
 
+  def archive
+    unless accessed?
+      flash.now.notice = 'Access denied!.'
+      return render :notification
+    end
```

Comment:
it should just raise active record not found, i.e. fix the code so that it finds with "!" on a specific scope.